src/qml/filters/fisheye/ui.qml

```diff
@@ -0,0 +1,827 @@
+            id: focalRatioSlider
+            minimumValue: 0
+            maximumValue: 1
+            decimals: 3
+            suffix: ''
+            value: focalRatioDefault
+            onValueChanged: { if (blockUpdate) return; filter.set(focalRatioParam, value) }
+        }
+        Shotcut.UndoButton {
+            onClicked: focalRatioSlider.value = focalRatioDefault
+        }
+
+        // Row 4: Interpolation Quality
+        Label {
+            text: qsTr('Quality')
+            Shotcut.HoverTip { text: qsTr( 'Resample quality') }
+            Layout.alignment: Qt.AlignRight
+        }
+        Shotcut.ComboBox {
+            id: qualityCombo
+            implicitWidth: 120
+            model: ListModel {
+                id: qualityModel
+                ListElement { text: qsTr('1. Nearest neighbor'); value: 0.0 }
+                ListElement { text: qsTr('2. Bilinear'); value: 0.166 }
+                ListElement { text: qsTr('3. Bicubic smooth'); value: 0.333 }
+                ListElement { text: qsTr('4. Bicubic sharp'); value: 0.500 }
+                ListElement { text: qsTr('5. Spline 4x4'); value: 0.666 }
+                ListElement { text: qsTr('6. Spline 6x6'); value: 0.833 }
+                ListElement { text: qsTr('7. Lanczos 16x16'); value: 1.0 }
+            }
+            textRole: 'text'
+            onActivated: {
+                filter.set(qualityParam, qualityModel.get(currentIndex).value)
+            }
+        }
+         Shotcut.UndoButton {
+            onClicked: {
+                filter.set(qualityParam, qualityDefault)
+                qualityCombo.currentIndex = setDefaultIndex(qualityDefault, qualityCombo.count) - 1
+            }
+        }
+
+        // Row 5: Lens Type
+        Label {
+            text: qsTr('Lens')
+            Shotcut.HoverTip { text: qsTr('Select a lens distortion pattern that best matches your camera') }
+            Layout.alignment: Qt.AlignRight
+        }
+        RowLayout {
+            Shotcut.ComboBox {
+                id: lensCombo
+                implicitWidth: 120
+                model: ListModel {
+                    id: lensModel
+                    ListElement { text: qsTr('1. Equidistant'); value: 0.0 }
+                    ListElement { text: qsTr('2. Ortographic'); value: 0.333 }
+                    ListElement { text: qsTr('3. Equiarea'); value: 0.666 }
+                    ListElement { text: qsTr('4. Stereographic'); value: 1.0 }
+                }
+                textRole: 'text'
+                onActivated: {
+                    filter.set(lensTypeParam, lensModel.get(currentIndex).value)
+                }
+            }
+            CheckBox {
+                text: qsTr('Dynamic stretch') //stretch view
+                leftPadding: 6
+                Shotcut.HoverTip { text: qsTr(  'Using a simplified version of \'Elastic Scale\' image will be stretched/squished to fix camera scaling between 4:3 and 16:9\n'+

```

Comment:
I prefer not to mention Elastic Scale here. You can simply replace "Using a simplified version of \'Elastic Scale\'" with "The" as those words do not add much and makes the person have to see what that is all about.